mobly/base_test.py

```diff
@@ -720,6 +720,7 @@ def should_retry(record):
       retry_name = f'{test_name}_retry_{i+1}'
```

Comment:
Trying to discuss here.

Should we still generate new names, now that the record clearly marks the sequence.
I guess it's easy for humans to understand.
Does the downstream result process also try to add the postfixes?